src/Features/LanguageServer/Protocol/Handler/Diagnostics/DocumentPullDiagnosticHandler.cs

```diff
@@ -76,14 +77,16 @@ protected override ImmutableArray<Document> GetOrderedDocuments(RequestContext c
             return ImmutableArray.Create(context.Document);
         }
 
-        protected override Task<ImmutableArray<DiagnosticData>> GetDiagnosticsAsync(
+        protected override async Task<ImmutableArray<DiagnosticData>> GetDiagnosticsAsync(
             RequestContext context, Document document, Option2<DiagnosticMode> diagnosticMode, CancellationToken cancellationToken)
         {
             // For open documents, directly use the IDiagnosticAnalyzerService.  This will use the actual snapshots
             // we're passing in.  If information is already cached for that snapshot, it will be returned.  Otherwise,
             // it will be computed on demand.  Because it is always accurate as per this snapshot, all spans are correct
             // and do not need to be adjusted.
-            return _analyzerService.GetDiagnosticsAsync(document.Project.Solution, documentId: document.Id, cancellationToken: cancellationToken);
+            var text = await document.GetTextAsync(cancellationToken).ConfigureAwait(false);
+            return await _analyzerService.GetDiagnosticsForSpanAsync(
+                document, new TextSpan(0, text.Length), cancellationToken: cancellationToken).ConfigureAwait(false);
```

Comment:
@mavasani  i have tested and validated this addresses teh perf problem we were seeing where the previous call would always end up causing us to compute diagnostics for hte entire project.  I initially was trying out the exercise of getting the initial call to not do that, effectively threading along the docId all the way and eventually getting to the point where i was about to make a DocumentAnalysisScope and use that to analyze the doc in question.

However, i wanted to see how this was used elsewhere and i saw that it was used if you happen to call this particular overload.  So i thought it woudl simpler and saner to just switch to calling this guy.  

This seems to work, but i'm very wary.  As before, i really don't know the contract of the analyzer service and how this may or may not end up working in practice.  It seems sane to do this, but i'm worried about unintended consequences of this.

To you have any thoughts on how we should proceed here?